provider/pkg/openapi/openapi.go

```diff
@@ -105,19 +105,19 @@ func PatchForResourceUpdate(
 	if knownGV := kinds.KnownGroupVersions.Has(lastSubmitted.GetAPIVersion()); !knownGV {
 		// Use a JSON merge patch for CRD Kinds.
 		patch, patchType, err = MergePatch(
-			lastSubmitted, lastSubmittedJSON, currentSubmittedJSON, liveOldJSON,
+			liveOldObj, lastSubmittedJSON, currentSubmittedJSON, liveOldJSON,
```

Comment:
The `MergePatch` and `StrategicMergePatch` methods take a `liveOld` parameter to obtain the object's GVK and name.  It doesn't affect the patch calculation, which is based on the `liveOldJSON` parameter.